SwanCustomEnvironments/swancustomenvironments/scripts/makenv.sh

```diff
@@ -189,14 +189,18 @@ if [ -n "${INSTALL_NXCALS}" ]; then
     # - PYSPARK_PYTHON: needed to point to the Python executable in the environment shipped to the cluster
     # - PATH: needed to point to all the binaries in the environment
     # - VIRTUAL_ENV: needed to point to the Python in the environment
+    # - SPARK_DRIVER_EXTRA_JAVA_OPTIONS: needed for gathering extraJavaOptions from the Spark default configuration, so it doesn't get overwritten by the SparkConnector
     export SPARK_HOME="${ENV_PATH}/nxcals-bundle"
     export PYSPARK_PYTHON="./environment/bin/python"
+    export SPARK_DRIVER_EXTRA_JAVA_OPTIONS=$(awk '/^spark.driver.extraJavaOptions/ {sub(/^spark.driver.extraJavaOptions /, ""); print}' ${SPARK_HOME}/conf/spark-defaults.conf)
```

Comment:
I believe we should place this variable definition in the acc-py.sh script, protected by an if so that it only gets done if NXCALS.

Reply:
Clarified, ignore.